### ISSUE TITLE
depth output not working when using OrthographicCamera

### ISSUE THREAD

**Issue by miccio-dk** [mmatl/pyrender#72]
Hello there,

I'm trying to obtain a depth map using the orthographic camera.
I took [this example code](https://pyrender.readthedocs.io/en/latest/examples/quickstart.html#minimal-example-for-offscreen-rendering) and replaced `PerspectiveCamera(yfov=np.pi / 3.0, aspectRatio=1.0)` with `OrthographicCamera(xmag=0.2,ymag=0.2)`.
This is what I get  with PerspectiveCamera:
![with PerspectiveCamera](https://user-images.githubusercontent.com/10885870/65053619-8013fe00-d96c-11e9-8a03-4bc1d1163c9f.png)

...and with OrthographicCamera:
![with OrthographicCamera](https://user-images.githubusercontent.com/10885870/65053454-4511ca80-d96c-11e9-8a71-d959051245bb.png)

It seems like all the geometry has disappeared and all I'm left with, is a dark outline.
In order to confirm this, I extracted the histogram of all the data in the depth channel:
<img width="639" alt="Screenshot 2019-09-17 17 02 22" src="https://user-images.githubusercontent.com/10885870/65053875-f3b60b00-d96c-11e9-8cdc-98efd7c62145.png">

Am I doing anything wrong?
Thanks in advance :)

**Comment by andreanicastro** [comment]
Hi there, 
I had the same issue and fixed it by patching the original code 

https://github.com/mmatl/pyrender/compare/master...andreanicastro:master

This is my patch

**Comment by miccio-dk** [comment]
Grazie Andrea! 
Just out of curiosity, do you know if the lib author is interested in merging your fix?

**Comment by ssavary** [comment]
I was coming here for the same issue. Nice work. Unless I misread it, I think you missed another place where your patch would required: read_depth_buf() in renderer.py

**Comment by andreanicastro** [comment]
Hi all,
I had a PR opened for two months and I was updating it to keep it in line with the repo. I closed it because I thought was not going to be merged. I can re-do the PR early next week :)

On 20 Sep 2019 19:27, Simon Savary <notifications@github.com> wrote:

I was coming here for the same issue. Nice work. Unless I misread it, I think you missed another place where your patch would required: read_depth_buf() in renderer.py

**Comment by andreanicastro** [comment]
@ssavary  You are right, I will have a look a the issue this evening. I think I will have a PR ready by tonight.

**Comment by muxizju** [comment]
> Hi there,
> I had the same issue and fixed it by patching the original code
> 
> [master...andreanicastro:master](https://github.com/mmatl/pyrender/compare/master...andreanicastro:master)
> 
> This is my patch

Im wondering why the author not merge the code in, it solves my issue

**Comment by lkskstlr** [comment]
@andreanicastro still very helpful, thank you!

**Comment by HugoFeng** [comment]
encontering this issue, thank you! don't know why it's still not fixed...

**Comment by kitizz** [comment]
For anyone else encountering this where patching `pyrender` is not an option, here's how to fix the output. I wasn't able to think of a reasonably robust check for general cases (in case future versions fix the bug). I've included the check that I know will work in my code because I've set `znear` and `zfar` to tight bounds that captures the full scene, and have a large spread relative to the buggy depths.

The equation below comes from rearranging the equations in @andreanicastro 's patch.

```python
color, depth = renderer.render(scene)
noninf = depth > 0
dmin = depth[noninf].min()
dmax = depth[noninf].max()

if (dmax - dmin) < (zfar - znear) / 10:  # CHOOSE A ROBUST CHECK FOR YOU
    # A hack to fix depth maps for OrthographicCamera.
    # See: https://github.com/mmatl/pyrender/issues/72
    depth[noninf] = zfar + znear - zfar * znear / depth[noninf]
```
